api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java

```diff
@@ -0,0 +1,170 @@
+package org.pmiops.workbench.db.dao;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.pmiops.workbench.compliance.ComplianceService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.firecloud.model.NihStatus;
+import org.pmiops.workbench.moodle.model.BadgeDetails;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.test.Providers;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.inject.Provider;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Random;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@Import({LiquibaseAutoConfiguration.class})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@AutoConfigureTestDatabase(replace= AutoConfigureTestDatabase.Replace.NONE)
+public class UserServiceTest {
+
+  private final String EMAIL_ADDRESS = "abc@fake-research-aou.org";
+
+  private Long incrementedUserId = 1L;
+
+  // An arbitrary timestamp to use as the anchor time for access module test cases.
+  private static final long TIMESTAMP_MSECS = 100;
+  private static final FakeClock CLOCK = new FakeClock(Instant.ofEpochMilli(TIMESTAMP_MSECS));
+
+  @Autowired
+  private UserDao userDao;
+  @Mock
+  private AdminActionHistoryDao adminActionHistoryDao;
+  @Mock
+  private FireCloudService fireCloudService;
+  @Mock
+  private ComplianceService complianceService;
+  private UserService userService;
+
+  private User testUser;
+
+  @Before
+  public void setUp() {
+    Provider<WorkbenchConfig> configProvider = Providers.of(WorkbenchConfig.createEmptyConfig());
+    testUser = insertUser(EMAIL_ADDRESS);
+
+    userService = new UserService(Providers.of(testUser), userDao, adminActionHistoryDao, CLOCK,
+        new Random(), fireCloudService, configProvider, complianceService);
+  }
+
+  @After
+  public void tearDown() {
+    userDao.deleteAll();
```

Comment:
This shouldn't be necessary based on my understanding of how spring data jpa testing works. If it is, please let me know what issue this is working around

Reply:
Yep – just leftover from prior futzing around with the tests. Removed.